Resources/Private/JavaScript/Host/Containers/LeftSideBar/ContentTree/index.js

```diff
@@ -0,0 +1,79 @@
+import React, {Component, PropTypes} from 'react';
+import Immutable from 'immutable';
+import {Tree} from 'Host/Components/';
+import {immutableOperations} from 'Shared/Util';
+import backend from 'Host/Service/Backend.js';
+import {connect} from 'react-redux';
+
+const {$immutable, $get, $set} = immutableOperations;
+
+
+let buildTreeData;
+buildTreeData = (nodes, nodePaths) => {
```

Comment:
Did you think of using reselect selector for this? Would improve performance and keep concerns a bit more separated. You can think of reselect as of Q in CQRS.